tests/compiletest.rs

```diff
@@ -41,317 +37,23 @@ fn run_tests(mode: Mode, path: &str, target: &str) {
-                    Error::OutputDiffers { path, actual, expected } =>
-                        compare_output(path, actual, expected),
-                }
-            }
-        }
-        eprintln!();
-        eprintln!(
-            "{} tests failed, {} tests passed, {} skipped",
-            failures.len().to_string().red().bold(),
-            (total - failures.len() - skipped).to_string().green(),
-            skipped.to_string().yellow()
-        );
-        std::process::exit(1);
-    }
-    eprintln!();
-    eprintln!(
-        "{} tests passed, {} skipped",
-        (total - skipped).to_string().green(),
-        skipped.to_string().yellow()
-    );
-}
-
-#[derive(Debug)]
-enum Error {
-    /// Got an invalid exit status for the given mode.
-    ExitStatus(Mode, ExitStatus),
-    PatternNotFound(String),
-    InlinePatternNotFound(String),
-    NoPatternsFound,
-    OutputDiffers {
-        path: PathBuf,
-        actual: String,
-        expected: String,
-    },
-}
-
-type Errors = Vec<Error>;
-
-fn revisions(path: &Path) -> Vec<String> {
-    let content = std::fs::read_to_string(path).unwrap();
-    for line in content.lines() {
-        if let Some(revisions) = line.strip_prefix("// revisions:") {
-            return revisions.trim().split_whitespace().map(|s| s.to_string()).collect();
-        }
-    }
-    vec![String::new()]
-}
-
-fn run_test(
-    path: &Path,
-    target: &str,
-    flags: &[String],
-    mode: Mode,
-    revision: &str,
-) -> (Command, Errors) {
-    // Run miri
-    let mut miri = Command::new(miri_path());
-    miri.args(flags.iter());
-    miri.arg(path);
-    if !revision.is_empty() {
-        miri.arg(format!("--cfg={revision}"));
-    }
-    miri.env("RUSTC_BACKTRACE", "0");
-    extract_env(&mut miri, path);
-    let output = miri.output().expect("could not execute miri");
-    let mut errors = mode.ok(output.status);
-    // Check output files (if any)
-    let revised = |extension: &str| {
-        if revision.is_empty() {
-            extension.to_string()
-        } else {
-            format!("{}.{}", revision, extension)
-        }
-    };
-    let stderr = check_output(&output.stderr, path, &mut errors, revised("stderr"), target);
-    check_output(&output.stdout, path, &mut errors, revised("stdout"), target);
-    let require = match mode {
-        Mode::Pass => false,
-        Mode::Panic => false, // Should we do anything here?
-        Mode::UB => true,
+    if let Some(target) = &target {
+        flags.push("--target".to_string());
+        flags.push(target.clone());
+    }
+
+    let config = Config {
+        args: flags,
+        target,
+        stderr_filters: REGEXES.clone(),
+        stdout_filters: REGEXES.clone(),
+        root_dir: PathBuf::from(path),
+        mode,
+        program: miri_path(),
+        bless: env::var_os("MIRI_BLESS").is_some(),
+        skip_output_checks: env::var_os("MIRI_SKIP_UI_CHECKS").is_some(),
```

Comment:
Please document these env vars in the README as well, like we do with all the others.